ci-operator/step-registry/upi/install/azure/upi-install-azure-commands.sh

```diff
@@ -198,10 +223,266 @@ az deployment group create -g $RESOURCE_GROUP \
   --parameters baseName="$INFRA_ID"
 
 echo "Deploying 03_infra"
-az deployment group create -g $RESOURCE_GROUP \
-  --template-file "03_infra.json" \
-  --parameters privateDNSZoneName="${CLUSTER_NAME}.${BASE_DOMAIN}" \
-  --parameters baseName="$INFRA_ID"
+if [ -v vnet_basename ] && [ -n "${vnet_basename}" ]; then
+  echo "Using an existing vnet, so use a customized infra file"
+  cat > infra_file.json << EOF  
```

Comment:
We cannot embed these files in the scripts like this. It makes them almost unreadable. We need to find another solution.

Reply:
How about putting these files in https://github.com/openshift/installer/tree/master/upi/azure, do you think that sounds reasonable?

Reply:
or shall I update the existing templates to introduce a new optional parameter - `vnet_basename` for that?

Reply:
> or shall I update the existing templates to introduce a new optional parameter - vnet_basename for that?

This sounds like the better option. Including a separate set of templates in the repo may be confusing.

Reply:
@patrickdillon submit https://github.com/openshift/installer/pull/5880 for it.